server/src/Main.hs

```diff
@@ -141,40 +169,41 @@ completeLogin origin rpIdHash db pending = do
     Right result -> pure result
 
   options <-
-    liftIO (getPendingOptions pending cred) >>= \case
+    Scotty.liftAndCatchIO (getPendingOptions pending cred) >>= \case
       Left err -> Scotty.raiseStatus HTTP.status401 $ "Challenge error: " <> LText.pack err
       Right result -> pure result
 
-  mentry <- liftIO $
+  mentry <- Scotty.liftAndCatchIO $
     Database.withTransaction db $ \tx ->
       Database.getCredentialEntryById tx (M.pkcIdentifier cred)
   entry <- case mentry of
     Nothing -> fail "Credential not found"
     Just entry -> pure entry
 
-  _newSigCount <- case verifyAssertionResponse origin rpIdHash Nothing entry options cred of
+  _newSigCount <- case verifyAssertionResponse origin rpIdHash (Just (ceUserHandle entry)) entry options cred of
     Failure (err :| _) -> fail $ show err
     Success result -> pure result
   -- FIXME: Update signature count in database
```

Comment:
Do we want to FIX this in a later PR?

Also, we can probably result in all errors, not just the first.

Reply:
Yeah, fixing that later :)

Reply:
And about showing all errors instead of just the first?

Reply:
Yeah that as well